### PR TITLE
fix: Java file diagnostics is not refreshed when properties file is saved

### LINKED ISSUE #860: Java file diagnostics is not refreshed when properties file is saved
See https://github.com/OpenLiberty/liberty-tools-intellij/issues/328

IBM did a fix fir this issue https://github.com/OpenLiberty/liberty-tools-intellij/pull/385 but it breaks LSP support.

As we have LSP console soon it will be easy to compare LSP notifications request responses with vscode

### PR DESCRIPTION
**Pull request by angelozerr**
fix: Java file diagnostics is not refreshed when properties file is saved

Fixes #860

### CODE REVIEW

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/AbstractConfigSource.java [hunk 1]
```diff
@@ -14,11 +14,16 @@
 import com.intellij.openapi.roots.ModuleRootManager;
 import com.intellij.openapi.vfs.VfsUtil;
 import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiFile;
+import io.quarkus.deployment.util.IoUtil;
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/AbstractConfigSource.java:18]: unused

File: src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/documentation/LSPDocumentationProvider.java [hunk 2]
```diff
@@ -100,9 +100,9 @@ public List<MarkupContent> getMarkupContents(PsiElement element, @Nullable PsiEl
             return ((LSPPsiElementForLookupItem) element).getDocumentation();
         }
         if (originalElement == null || !Objects.equals(element.getContainingFile(), originalElement.getContainingFile())) {
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/documentation/LSPDocumentationProvider.java:102]: might as well remove the if block

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/AbstractConfigSource.java (hunk 1 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/AbstractConfigSource.java]: fixed

File: src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/documentation/LSPDocumentationProvider.java (hunk 2 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4ij/operations/documentation/LSPDocumentationProvider.java]: my bad -( fixed

**Comment by fbricon** [comment]
Works great. Let's merge after CI passes

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java [hunk 3]
```diff
@@ -1,272 +1,291 @@
...
-			int intValue = Integer.parseInt(value);
-			return intValue;
-		} catch (NumberFormatException nfe) {
-			return defaultValue;
-		}
-	}
+    /**
+     * Returns the value of this property as an int, or <code>defaultValue</code>
+     * when there is no value or the value cannot be parsed as a String.
+     * <p>
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java:101]: missing closing tag

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java [hunk 4]
```diff
@@ -1,272 +1,291 @@
...
-	 * @param defaultValue the value to return if the value for the property is not
-	 *                     defined in this project
-	 * @return the value of this property or <code>defaultValue</code> if it is not
-	 *         defined in this project
-	 */
-	public String getProperty(String propertyKey, String defaultValue) {
+    /**
+     * Returns the value of this property or <code>defaultValue</code> if it is not
+     * defined in this project.
+     * <p>
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java:48]: missing closing tag

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java [hunk 5]
```diff
@@ -1,272 +1,291 @@
...
-	 * @return the value of this property or null if it is not defined in this
-	 *         project
-	 */
-	public String getProperty(String propertyKey) {
-		return getProperty(propertyKey, null);
-	}
+    /**
+     * Returns the value of this property or null if it is not defined in this
+     * project.
+     * <p>
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java:85]: missing closing tag

**Review by fbricon — CHANGES_REQUESTED** [review]

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java (hunk 5 above)
Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java:86]: fixed

File: src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java [hunk 6]
```diff
@@ -1,272 +1,302 @@
...
-	/**
-	 * Evict the config sources cache and related cached information as soon as one
-	 * of properties, yaml file is saved.
-	 */
-	public void evictConfigSourcesCache() {
-		configSources = null;
-		propertyValueExpander = null;
-		aggregatedPropertiesProvider = null;
-	}
+    private IConfigSource findCounfigSource(VirtualFile file) {
```
Comment (fbricon) [inline src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java:219]: findConfigSource

Comment (angelozerr) [reply src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/project/PsiMicroProfileProject.java]: fixed

**Comment by angelozerr** [comment]
>Works great. Let's merge after CI passes

CI are passed

**Comment by fbricon** [comment]
Thanks @angelozerr

**Pull request merged**